app/src/main/java/emu/skyline/MainViewModel.kt

```diff
@@ -73,4 +73,20 @@ class MainViewModel @Inject constructor(@ApplicationContext context : Context, p
             }
         }
     }
+
+    /**
+     * This checks if the roms have changed since the last time they were loaded and if so it reloads them
+     */
+    fun checkRomHash(context : Context, searchLocation : Uri, systemLanguage : Int) {
+        CoroutineScope(Dispatchers.IO).launch {
+            val oldHashFile = File(getApplication<SkylineApplication>().filesDir.canonicalPath + "/roms.txt")
+            val oldHash = if (oldHashFile.exists()) oldHashFile.readText().toIntOrNull() else null
+            val romElements = romProvider.loadRoms(searchLocation, systemLanguage).values.flatten().toList()
+            val newHash = romElements.map { it.name }.toSet().hashCode()
+            if (newHash != oldHash) {
+                loadRoms(context, false, searchLocation, systemLanguage)
```

Comment:
Since you've already have loaded the new ROMs, there's no need to request a new load, you can just assign them here. Since this is a `LiveData` object, changes will be automatically propagated to the game view.
```suggestion
                state = MainState.Loaded(romElements)
```

Reply:
👍